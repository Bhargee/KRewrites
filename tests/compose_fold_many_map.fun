fold (fun a b -> a + b) map (fun c -> c + 1) map (fun d -> d -1)
