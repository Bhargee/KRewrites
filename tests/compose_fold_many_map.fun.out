Search results:

Solution 1:
<T>
    <k>
        fold ( fun (a b) -> (a + b) ) -| ( ( fun c -> (c + 1) ) ) -| ( ( 
          fun d -> (d - 1) ) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 2:
<T>
    <k>
        fold ( fun (a b) -> (a + b) ) -| ( ( fun c -> (c + 1) ) ) -| ( fun
           d -> (d - 1) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 3:
<T>
    <k>
        fold ( fun (a b) -> (a + b) ) -| ( fun c -> (c + 1) ) -| ( ( fun d
           -> (d - 1) ) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 4:
<T>
    <k>
        fold ( fun (a b) -> (a + b) ) -| ( fun c -> (c + 1) ) -| ( fun d 
          -> (d - 1) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 5:
<T>
    <k>
        fold ( fun c -> (c + 1) ) -| ( ( fun (a b) -> (a + b) ) ) -| ( ( 
          fun d -> (d - 1) ) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 6:
<T>
    <k>
        fold ( fun c -> (c + 1) ) -| ( ( fun (a b) -> (a + b) ) ) -| ( fun
           d -> (d - 1) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 7:
<T>
    <k>
        fold ( fun c -> (c + 1) ) -| ( fun (a b) -> (a + b) ) -| ( ( fun d
           -> (d - 1) ) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 8:
<T>
    <k>
        fold ( fun c -> (c + 1) ) -| ( fun (a b) -> (a + b) ) -| ( fun d 
          -> (d - 1) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 9:
<T>
    <k>
        fold ( fun d -> (d - 1) ) -| ( ( fun (a b) -> (a + b) ) -| ( ( fun
           c -> (c + 1) ) ) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 10:
<T>
    <k>
        fold ( fun d -> (d - 1) ) -| ( ( fun (a b) -> (a + b) ) -| ( fun c
           -> (c + 1) ) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 11:
<T>
    <k>
        fold ( fun d -> (d - 1) ) -| ( ( fun c -> (c + 1) ) -| ( ( fun (a 
          b) -> (a + b) ) ) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>

Solution 12:
<T>
    <k>
        fold ( fun d -> (d - 1) ) -| ( ( fun c -> (c + 1) ) -| ( fun (a b)
           -> (a + b) ) )
    </k>
    <store>
        .Map
    </store>
    <env>
        .Map
    </env>
</T>
